src/isl/isl_type_reference.rs

```diff
@@ -103,45 +84,98 @@ impl IslTypeRef {
         }
     }
 
-    // TODO: break match arms into helper methods as we add more constraints
-    /// Resolves a type_reference into a [TypeId] using the type_store
-    pub fn resolve_type_reference(
-        type_reference: &IslTypeRef,
+    /// Return TypeId for given built-in type name from type_store
+    fn get_type_id_from_built_in_type_name(
+        alias: &str,
         type_store: &mut TypeStore,
         pending_types: &mut PendingTypes,
     ) -> IonSchemaResult<TypeId> {
-        match type_reference {
-            IslTypeRef::CoreIsl(ion_type) => {
-                // TODO: create CoreType struct for storing ISLCoreType type definition instead of Type
-                // inserts ISLCoreType as a Type into type_store
-                Ok(pending_types.add_core_type(ion_type.to_owned(), type_store))
-            }
-            IslTypeRef::Named(alias) => {
+        use IonType::*;
+        let invalid_type_reference_error = unresolvable_schema_error_raw(format!(
```

Comment:
question—How do you decide when to use `unresolvable_schema_error_raw` vs `unresolvable_schema_error`?

Reply:
Basically `unresolvable_schema_error_raw` returns `IonSchemaError` whereas `unresolvable_schema_error` returns `IonSchemaResult` which is `IonSchemaError` wrapped inside a `Result`.